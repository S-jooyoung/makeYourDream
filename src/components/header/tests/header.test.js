// import dependencies
import React from "react";

// import react-testing methods
import { render, screen } from "@testing-library/react";

import Header from "../header";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";

describe("Header", () => {
  let HeaderCompoent;
  let onLogout;

  beforeEach(() => {
    onLogout = jest.fn();
    HeaderCompoent = <Header onLogout={onLogout} />;
  });

  it("render", () => {
    const component = renderer.create(HeaderCompoent).toJSON();

    expect(component).toMatchSnapshot();
  });

  describe("onLogout", () => {
    beforeEach(() => {
      render(HeaderCompoent);
    });

    it('calls onLogout when clicking "로그아웃" button', () => {
      let button = screen.getByTitle("button");
      userEvent.click(button);

      expect(onLogout).toHaveBeenCalled();
    });
  });
});
