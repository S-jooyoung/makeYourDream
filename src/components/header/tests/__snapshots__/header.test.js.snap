// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render 1`] = `
<header
  className="header"
>
  <div
    className="logoutBtn"
  >
    <button
      className="button"
      onClick={[MockFunction]}
      title="button"
    >
      로그아웃
    </button>
  </div>
  <img
    alt="logo"
    className="logo"
    src="/images/loginLogo.png"
  />
</header>
`;
